File: node/lib/Customer.js
///<reference path='../typings/lodash/lodash.d.ts'/>
'use strict';
var _ = require('lodash');

function defineWritableProperties(instance, properties) {
  Object.defineProperties(instance, _.reduce(properties, function (memo, next) {
    memo[next] = {
      value: null,
      writable: true
    };
    return memo;
  }, {}));
}

function Customer() {
  var properties = [
    'accountNumber',
    'addressCity',
    'addressCountry',
    'addressStreet',
    'firstName',
    'gender',
    'lastName',
    'pictureUri'];
  defineWritableProperties(this, properties);
  Object.seal(this);
}
module.exports = Customer;

File: clean-aspnet-core/CustomerService/Api/Customer.cs
using System;
using System.Runtime.Serialization;
using System.Xml.Serialization;
namespace Customers
{
    public partial class Customer : object
    {
        public int AccountNumber{ get; set; }
        public string AddressCity{ get; set; }
        public string AddressCountry{ get; set; }
        public string AddressStreet{ get; set; }
        public string FirstName{ get; set; }
        public CustomerGender Gender{ get; set; }
        public string LastName{ get; set; }
        public Uri PictureUri{ get; set; }
    }
}